File: Android/java/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/datasource/RamDataSource.java
package com.didichuxing.doraemonkit.kit.performance.datasource;

import com.didichuxing.doraemonkit.kit.performance.PerformanceDataManager;
import com.didichuxing.doraemonkit.kit.performance.widget.LineData;

public class RamDataSource implements IDataSource {
    private float mMaxRam;

    public RamDataSource() {
        mMaxRam = (float) (Runtime.getRuntime().maxMemory() * 1.0 / (1024 * 1024));
    }

    @Override
    public LineData createData() {
        float info = PerformanceDataManager.getInstance().getLastMemoryInfo();
        return LineData.obtain(info / mMaxRam * 100, Math.round(info) + "MB");
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/manager/datasource/RamDataSource.kt
package com.didichuxing.doraemonkit.kit.performance.manager.datasource

import com.didichuxing.doraemonkit.kit.performance.manager.PerformanceDataManager
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData.Companion.obtain
import kotlin.math.roundToInt

/**
 *
 * Desc:内存数据工厂实现类
 * <p>
 * Date: 2020-06-09
 * Updater:
 * Update Time:
 * Update Comments:
 *
 * Author: pengyushan
 */
class RamDataSource : IDataSource {
    private val mMaxRam: Float = (Runtime.getRuntime().maxMemory() * 1.0 / (1024 * 1024)).toFloat()
    override fun createData(): LineData? {
        val info: Float = PerformanceDataManager.instance.lastMemoryInfo
        return obtain(info / mMaxRam * 100, info.roundToInt().toString() + "MB")
    }

}